Code/6-wire-version/src/main.cpp

```diff
@@ -21,78 +21,113 @@ void setup() {
 			// int dsp_audio_pin 	= D6 
 			// );
 	//example: bwc.begin(D1, D2, D3, D4, D5, D6, D7);
-  updateMqttTimer.attach(600, sendMQTTsetFlag);     //update mqtt every 10 minutes. Mqtt will also be updated on every state change
-  updateWSTimer.attach(2.0, sendWSsetFlag);         //update webpage every 2 secs plus state changes
-  periodicTimer.attach(60, periodicTimerCallback);  //things to do once in a while
-  LittleFS.begin(); // needs to be loaded here for reading the wifi.json
+
+  // check things every 60 seconds
+  periodicTimer.attach(60, []{ periodicTimerFlag = true; });
+  
+  // update webpage every 2 seconds. (will also be updated on state changes)
+  updateWSTimer.attach(2.0, []{ sendWSFlag = true; });
+
+  // update MQTT every 10 minutes. (will also be updated on state changes)
+  updateMqttTimer.attach(600, []{ sendMQTTFlag = true; });
+
+  // needs to be loaded here for reading the wifi.json
+  LittleFS.begin();
   loadWifi();
   startWiFi();
   startNTP();
   startOTA();
-  startServer();
+  startHttpServer();
   startWebSocket();
   startMQTT();
+
   bwc.print(WiFi.localIP().toString());
-  Serial.println(F("\nEnd of setup()"));
+  Serial.println(F("End of setup()"));
 }
 
-void loop() {
-  bwc.loop();                     // Fiddle with the pump computer
-  if(WiFi.status() == WL_CONNECTED){
-    webSocket.loop();             // check for websocket events
-    server.handleClient();        // run the server
-    ArduinoOTA.handle();          // listen for OTA events
-      
-    if (enableMqtt){
-      if(!MQTTclient.loop()){
+void loop()
+{
+  // Fiddle with the pump computer
+  bwc.loop();
+
+  // do things only when WiFi is connected
+  if (WiFi.status() == WL_CONNECTED)
+  {
+    // listen for websocket events
+    webSocket.loop();
+    // listen for webserver events
+    server.handleClient();
+    // listen for OTA events
+    ArduinoOTA.handle();
+    
+    // MQTT
+    if (enableMqtt)
+    {
+      if (!MQTTclient.loop())
+      {
         MQTT_Connect();
```

Comment:
If connection fails this will bog down the ESP. I tried to only connect every 60 s but it didn't connect at all :-(